mesonbuild/dependencies/boost.py

```diff
@@ -152,7 +159,7 @@ def __init__(self, path: Path):
         elif self.nvsuffix in ['a', 'lib']:
             self.static = True
         else:
-            raise DependencyException(f'Unable to process library extension "{self.nvsuffix}" ({self.path})')
+            raise UnknownFileException(self.path)
```

Comment:
Also, there should be a warning (or at least a debug log) here to indicate that there was an unexpected file. This way, debugging further issues will be easier since there is a specific reason printed in the `meson-log.txt`.